### PR TITLE
Enable linting on the repository

### LINKED ISSUE #18: Add linting to the workflow and fix any issues found by the linter

### PR DESCRIPTION
**Pull request by aakselrod**
Fixes #18. Based on #11 and should be rebased after it's merged.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,6 +21,7 @@ RUN cd /root && \
     unzip vault_${vault}_linux_${goplatform}.zip && \
     mv vault /usr/local/bin/ && \
     go install github.com/go-delve/delve/cmd/dlv@latest && \
+    go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest && \
```
Comment (joostjager) [inline Dockerfile]: Nice that the same image can be used for linting.

There is also a golangci-lint github action available though. See for example https://github.com/lightningequipment/circuitbreaker/blob/master/.github/workflows/golangci-lint.yml

**Review by joostjager — COMMENTED** [review]
To minimize overall delta, it would probably be better to first enable the linter and only then merge #11 without linter issues. But if for practical reasons you want to do it the other way around, that's also fine.

File: Dockerfile (hunk 1 above)
Comment (aakselrod) [reply Dockerfile]: I've reworked this using a github action, and rebased on master so that I can base #11 on top of this PR instead.

**Comment by aakselrod** [comment]
Thanks for the review! Since I have a bit of rework to do in #11, I think we can try to get this in first, and the rest of the lint fixes can go into their origins in #11 directly.

**Review by joostjager — APPROVED** [review]
Nice finds by the linter.

For uber-cleanliness you could first fix the linter problems in commits and only then enable the linter. But ofc the end result is the same.

**Comment by aakselrod** [comment]
I've put the linter-enabling commit last to avoid any lint failures on the main branch before merge.

**Pull request merged**